inmoose/limma/ebayes.py

```diff
@@ -0,0 +1,161 @@
+# -----------------------------------------------------------------------------
+# Copyright (C) 2004-2022 Gordon Smyth, Yifang Hu, Matthew Ritchie, Jeremy Silver, James Wettenhall, Davis McCarthy, Di Wu, Wei Shi, Belinda Phipson, Aaron Lun, Natalie Thorne, Alicia Oshlack, Carolyn de Graaf, Yunshun Chen, Mette Langaas, Egil Ferkingstad, Marcus Davy, Francois Pepin, Dongseok Choi
+# Copyright (C) 2024 Maximilien Colange
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+# This file is based on the file 'R/ebayes.R' of the Bioconductor limma package (version 3.55.1).
+
+
+import numpy as np
+import scipy
+
+
+def tmixture_matrix(tstat, stdev_unscaled, df, proportion, v0_lim=None):
```

Comment:
Could you add variable type ?

Reply:
types are documented in the docstring
I'm not sure that type annotation would add much info

Reply:
It's just that when you use it outside and you have the type in the function definition you can see it easily. If you think it is not worth it I trust you